### PR TITLE
AVM2 TextField impl

### PR DESCRIPTION
**Pull request by kmeisthax**
This PR enables text fields to be created and accessed from AVM2.

 * [x] Create fresh text fields when `TextField` is constructed, and AVM2 objects for timeline-constructed fields
 * [x] Impl `TextFormat` & conversions to and from AVM2
 * [x] Impl `TextFieldAutoSize`
 * [x] Impl `TextFormatAlign`
 * [x] Impl `TextField` properties up to parity with AS2
 * [x] Port AS2 `edittext_*` tests to AS3 and pass them

This PR is not planned to implement *new* text field functionality, such as the new alignment options and properties not accessible from AVM2.

### CODE REVIEW

**Comment by nagatsu-chan** [comment]
I love you

**Comment by kmeisthax** [comment]
I'm releasing the draft lock. I was initially concerned about tests being off in AS3, but the factor that they're off is actually less than the AS2 ones. I can worry about that later.

In AS3 we actually have the tools to introspect text layout decisions (via `getLineMetrics`), so I can actually fix long-standing text layout bugs in future PRs. I'm not doing this here since this PR exists *only* to enable text fields on AS3.

**Review by Herschel — APPROVED** [review]

File: core/src/avm2/array.rs [hunk 1]
```diff
@@ -39,6 +39,18 @@ impl<'gc> ArrayStorage<'gc> {
         Self { storage }
     }
 
+    /// Convert any iterable stream of values or value-like items into array
+    /// storage.
+    pub fn from_iter<I, V>(values: I) -> Self
```
[the remark this answers is not in the document]
Comment (kmeisthax) [reply core/src/avm2/array.rs]: Turns out it can.

I remember it not being able to at some point, but maybe I just screwed up the trait bounds.

**Comment by Herschel** [comment]
Thank you!

**Pull request merged**